proposals/0371-isolated-synchronous-deinit.md

```diff
@@ -94,21 +94,28 @@ actor Clicker {
 }
 ```
 
-## Detailed design
-
-### Runtime
+Note that since Swift 5.8 escaping `self` from `deinit` [reliably triggers fatal error](https://github.com/apple/swift/commit/108f780).
 
-This proposal introduces a new runtime function that is used to schedule `deinit`'s code on an executor by wrapping it into a task-less ad hoc job. It does not do any reference counting and can be safely used even with references that were released for the last time but not deallocated yet.
+If needed, it is still possible to manually start a task from `deinit`. But any data needed by the task should be copied to avoiding capturing `self`:
 
-If no switching is needed, then `deinit` is executed immediately on the current thread. Otherwise, the task-less job copies priority and task-local values from the task/thread that released the last strong reference to the object.
+```swift
+actor Clicker {
+  var count: Int = 0
 
-```cpp
-using DeinitWorkFunction = SWIFT_CC(swift) void (void *);
-  
-SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
-void swift_task_deinitOnExecutor(void *object, DeinitWorkFunction *work, ExecutorRef newExecutor);
+  isolated deinit {
+    Task { [count] in 
+      await logClicks(count)
+    }
+  }
+}
 ```
 
+## Detailed design
+
+### Runtime
```

Comment:
Honestly, I think we could strike most of this section, it wades a bit heavily into implementation details that aren't necessary to understand the language model, I think. How about:
> This proposal introduces a new runtime function which ensures that the body of an `isolated deinit` is running on the correct executor by wrapping the execution in a task-less ad hoc job, if needed. If no switching is needed (i.e., we are already executing on the appropriate executor), then the `deinit` is executed synchronously. Otherwise, the task-less job is schedule with the same priority as the current task/thread that release the last strong reference to the object.